marklogic-data-hub-central/ui/src/components/flows/flows.tsx

```diff
@@ -557,33 +594,58 @@ const Flows: React.FC<Props> = (props) => {
     return disabledCheck;
   };
 
+  let titleTypeStep; let currentTitle = "";
+  let mapTypeSteps = new Map([["mapping", "Map"], ["merging", "Merge"], ["custom", "Custom"], ["mastering", "Master"], ["ingestion", "Load"]]);
+  const handleTitleSteps = (stepType) => {
+    if (currentTitle !== stepType) {
+      titleTypeStep = mapTypeSteps.get(stepType) ? mapTypeSteps.get(stepType) : "";
+    } else { titleTypeStep = ""; }
+
+    currentTitle = stepType;
+    return titleTypeStep;
+  };
+
   const flowMenu = (flowName) => {
     return (
       <>
-        <Dropdown.Header className="py-0 fs-6 mb-2">Select the steps to include in the run:</Dropdown.Header>
+        <Dropdown.Header className="py-0 fs-6 mb-2 text-dark">Select the steps to run in this flow:</Dropdown.Header>
+        <hr />
+        <div className={styles.divCheckAll}>
+          <HCCheckbox
+            id={"checkAll"}
+            value={checkAll}
+            checked={checkAll}
+            handleClick={(event) => onCheckboxChange(event, "", "", "", flowName, "", "", true)}
+            label={controlStepSelected(flowName) ? "Deselect All" : "Select All"}
+          >
+          </HCCheckbox>
+        </div>
         {props.flows.map((flow) => (
           flow["name"] === flowName &&
           flow.steps.map((step, index) => (
-            <div id={index} className={styles.divItem}>
-              <HCTooltip text={step.stepDefinitionType === "ingestion" ? controlDisabled(step, flowName) ? "Only 1 load step can be selected": "": ""} placement="left" id={`tooltip`}>
-                <div>
-                  <HCCheckbox
-                    id={step.stepName}
-                    value={step.stepName}
-                    handleClick={(event) => onCheckboxChange(event, step.stepName, step.stepNumber, step.stepDefinitionType, flowName, step.stepId, step.sourceFormat)}
-                    checked={selectedStepOptions[flowName + "_" + step.stepName + "_" + step.stepNumber] ? true:false}
-                    disabled={step.stepDefinitionType === "ingestion" ? controlDisabled(step, flowName) : false}
-                  >{step.stepName}
-                  </HCCheckbox></div></HCTooltip>
-            </div>
+            <>
+              <div className={styles.titleTypeStep}>{handleTitleSteps(step.stepDefinitionType)}</div>
+              <div id={index} className={styles.divItem}>
+                <HCTooltip text={step.stepDefinitionType === "ingestion" ? controlDisabled(step, flowName) ? "Only one Loading step can be included in a flow" : "" : ""} placement="left" id={`tooltip`}>
```

Comment:
Can we move the tooltips to a common component, like {RunTooltips} in "@config/tooltips.config"

Reply:
Done.